satellite/metainfo/metainfo.go

```diff
@@ -448,12 +451,14 @@ func createBucketID(projectID uuid.UUID, bucket []byte) []byte {
 	return []byte(storj.JoinPaths(entries...))
 }
 
-func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Pointer) (err error) {
+func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Pointer, limits []*pb.OrderLimit2) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
 	if pointer.Type == pb.Pointer_REMOTE {
 		var remotePieces []*pb.RemotePiece
 		remote := pointer.Remote
+		allSizesValid := true
+		maxPieceSize := int64(0)
```

Comment:
`maxPieceSize` is misleading var name. It looks like here you record the size of the first piece and then check if all other pieces have the same size.

Reply:
I cannot use first piece size because some pieces could be nil e.g first one. I will think to make more straightforward.

Reply:
renamed to `lastPieceSize` and removed condition to update it only with larger piece size